### PR TITLE
Adds support for use of enum types as literals and nativeEnums

### LINKED ISSUE #39: Falls back to z.any() for enum type literals and fails validation
## Bug description

When using a [discriminated union](https://basarat.gitbook.io/typescript/type-system/discriminated-unions) over enum types, `ts-to-zod` will fallback to using `z.any` instead of using the enum as a literal. 

```ts
// This works with strings 
type ExampleA {
  kind: 'a'
};

type ExampleB {
  kind: 'b',
  otherProps: string,
};

type Example = ExampleA | ExampleB;


// But fails with enums 

enum ExampleKind {
 A = "a",
 B = "b"
}

type ExampleA {
  kind: ExampleKind.A
};

type ExampleB {
  kind: ExampleKind.B
  otherProps: string,
};

type Example = ExampleA | ExampleB;
```

### Input

```ts

export enum TagNameKind {
	Button = 'button'
};

export type TagNameButton = {
	kind: TagNameKind.Button
};

export type TagName = TagNameButton;

export type Tag = {
	tag: TagName,
}

```

### Expected output

```ts
// Generated by ts-to-zod
import { z } from 'zod'
import { TagNameKind } from './../types/tag-simple'

export const tagNameKindSchema = z.nativeEnum(TagNameKind)

export const tagNameButtonSchema = z.object({
  kind: z.literal(TagNameKind.Button)
})

export const tagNameSchema = tagNameButtonSchema

export const tagSchema = z.object({
  tag: tagNameSchema
})

```

### Actual output

```ts
// Generated by ts-to-zod
import { z } from 'zod'
import { TagNameKind } from './../types/tag-simple'

export const tagNameKindSchema = z.nativeEnum(TagNameKind)

export const tagNameButtonSchema = z.object({
  kind: z.any()
})

export const tagNameSchema = tagNameButtonSchema

export const tagSchema = z.object({
  tag: tagNameSchema
})

```

Without passing `--skipValidation`

```
 »   Warning: 'TypeReference' is not supported, fallback into 'z.any()'
✖ Validating generated types
 ›   Error: 'tagNameButtonSchema' is not compatible with 'TagNameButton':
 ›   Argument of type '{ kind?: any; }' is not assignable to parameter of type 'TagNameButton'.
 ›     Property 'kind' is optional in type '{ kind?: any; }' but required in type 'TagNameButton'.
 ›   'tagNameSchema' is not compatible with 'TagName':
 ›   Argument of type '{ kind?: any; }' is not assignable to parameter of type 'TagNameButton'.
 ›   'tagSchema' is not compatible with 'Tag':
 ›   Argument of type '{ tag: { kind?: any; }; }' is not assignable to parameter of type 'Tag'.
 ›     Types of property 'tag' are incompatible.
 ›       Type '{ kind?: any; }' is not assignable to type 'TagNameButton'.
error Command failed with exit code 2.

```

**Versions**

- Typescript: `v4.3.4`
- Zod: `v3.1.0`

### PR DESCRIPTION
**Pull request by anglinb**
# Why

Fixes #39 and #35

### Warnings ⚠️ 
- ~Relies on https://github.com/fabien0102/ts-to-zod/pull/38~ Rolled these together b/c #38 was a subset of this PR
- Does not handle imports, assumes enum is exported & imported already

### CODE REVIEW

File: src/core/generateZodSchema.ts [hunk 1]
```diff
@@ -447,6 +455,25 @@ function buildZodPrimitive({
     }
   }
 
+  // Deal with enums used as literals
+  if (
```
Comment (fabien0102) [inline src/core/generateZodSchema.ts:459]: Let's add a small test in `generateZodSchema.test.ts`

**Review by fabien0102 — CHANGES_REQUESTED** [review]
Looks good, let's just add a little unit test to make sure everything works as expected

File: src/core/generateZodSchema.ts (hunk 1 above)
Comment (anglinb) [reply src/core/generateZodSchema.ts:458]: https://github.com/fabien0102/ts-to-zod/pull/40#discussion_r678971858

**Comment by anglinb** [comment]
I think I addressed all the feedback, thanks again for taking a look @fabien0102 🙇

**Review by fabien0102 — APPROVED** [review]
Looks good, nice job! I will publish a new release just after 😀

**Comment by dvv** [comment]
I wonder whether the following is a bug:

```
export enum LoggerLevel { debug, info, warning, error, critical }
interface ILogger {
  /**
   * @default LoggerLevel.info
   */
  level: LoggerLevel
}
```
compiles down to
```
export const loggerLevelSchema = z.nativeEnum(LoggerLevel);
const iLoggerSchema = z.object({
  level: loggerLevelSchema.default("LoggerLevel.info"), //// <----- notice enum default is quoted
});
```

TIA

**Pull request merged**